crates/consensus/src/transaction/envelope.rs

```diff
@@ -658,6 +658,67 @@ impl Typed2718 for TxEnvelope {
     }
 }
 
+/// Transaction envelope.
+#[auto_impl::auto_impl(&, Arc)]
+pub trait TransactionEnvelope: Typed2718 {
+    /// Returns ref to [`Signed<TxLegacy>`] if transaction is a legacy transaction.
+    fn as_legacy(&self) -> Option<&Signed<TxLegacy>>;
```

Comment:
this bakes in the signed type, which I think is fine because I think this will not be used by reth.

~~but the function name could be problematic, could we introduce `as_signed_*` instead?~~

ah I guess this isn't necessary:

https://github.com/alloy-rs/alloy/blob/f390d442b11a667f18c93cfff68ec081dab54c63/crates/consensus/src/transaction/envelope.rs#L282-L282

Reply:
I guess these constant functions are likely not needed anywhere